evm/evm_transaction_opts.go

```diff
@@ -53,19 +56,44 @@ func newTransactOptsBuilder(privKey *ecdsa.PrivateKey) transactOpsBuilder {
 	}
 }
 
+const (
+	MalleabilityThreshold = "0x7fffffffffffffffffffffffffffffff5d576e7357a4501ddfe92f46681b20a0"
+	ZeroSValue            = "0x0000000000000000000000000000000000000000000000000000000000000000"
+)
+
 // SigToVRS breaks apart a signature into its components to make it compatible with the contracts
-func SigToVRS(sigHex string) (v uint8, r, s ethcmn.Hash) {
-	signatureBytes := ethcmn.FromHex(sigHex)
+func SigToVRS(sigHex string) (v uint8, r, s ethcmn.Hash, err error) {
+	signatureBytes := ethcmn.FromHex(strings.ToLower(sigHex))
+
+	// signature length should be 65: 32 bytes + vParam
+	if len(signatureBytes) != 65 {
+		err = errors.Wrap(ErrInvalid, "signature length")
+		return
+	}
+
+	// vParam should be 0, 1, 27 or 28
 	vParam := signatureBytes[64]
-	if vParam == byte(0) {
+	switch vParam {
+	case byte(0):
 		vParam = byte(27)
-	} else if vParam == byte(1) {
+	case byte(1):
 		vParam = byte(28)
+	case byte(27):
```

Comment:
mind adding some reference for posterity here? I think there's some EIP or something? or an issue that explains it also works

Reply:
[d402e8a](https://github.com/celestiaorg/orchestrator-relayer/pull/201/commits/d402e8a31bd2d52e31fd70a86eb9a4703988ae6d)

Reply:
thanks!